src/rdkafka_partition.c

```diff
@@ -2865,6 +2865,21 @@ rd_kafka_topic_partition_list_add(rd_kafka_topic_partition_list_t *rktparlist,
 }
 
 
+rd_kafka_topic_partition_t *rd_kafka_topic_partition_list_add_with_topic_id(
+    rd_kafka_topic_partition_list_t *rktparlist,
+    rd_kafka_uuid_t topic_id,
```

Comment:
Merge `dev_kip848_topic_partition_and_error_codes` into this branch and rename `rd_kafka_uuid_t` to `rd_kafka_Uuid_t` before squash merging the other way around